pkg/pgclient/client.go

```diff
@@ -35,24 +35,34 @@ type LockFunc = func(ctx context.Context, conn *pgx.Conn) error
 
 // NewClient creates a new PostgreSQL client
 func NewClient(cfg *Config, schemaLocker LockFunc) (*Client, error) {
-	connectionStr := cfg.GetConnectionStr()
+	connectionStr, err := cfg.GetConnectionStr()
+	if err != nil {
+		return nil, err
+	}
 	minConnections, maxConnections, numCopiers, err := cfg.GetNumConnections()
 	if err != nil {
 		log.Error("msg", "configuring number of connections", "err", util.MaskPassword(err.Error()))
 		return nil, err
 	}
 
-	pgConfig, err := pgxpool.ParseConfig(connectionStr + fmt.Sprintf(" pool_max_conns=%d pool_min_conns=%d", maxConnections, minConnections))
-	if err != nil {
-		log.Error("msg", "configuring connection", "err", util.MaskPassword(err.Error()))
-		return nil, err
+	var pgConfig *pgxpool.Config
+	if cfg.DbUri == DefaultDBUri {
+		pgConfig, err = pgxpool.ParseConfig(connectionStr + fmt.Sprintf(" pool_max_conns=%d pool_min_conns=%d", maxConnections, minConnections))
+		if err != nil {
+			log.Error("msg", "configuring connection", "err", util.MaskPassword(err.Error()))
+			return nil, err
+		}
+	} else {
+		pgConfig, err = pgxpool.ParseConfig(connectionStr + fmt.Sprintf("&pool_max_conns=%d&pool_min_conns=%d", maxConnections, minConnections))
+		if err != nil {
+			log.Error("msg", "configuring connection", "err", util.MaskPassword(err.Error()))
+			return nil, err
+		}
```

Comment:
Would lessen the code duplication if you were to only construct the connection string in the conditional path. You might even want to just switch between the concatenation strings ` ` or `&` depending whether its a DB URI or not.

Should make the code simpler and easier to read.

Reply:
I will move to check for error redundant code out of if-else for more code singularity.